src/formio/components/partners.ts

```diff
@@ -0,0 +1,21 @@
+import {TableViewComponentSchema} from '..';
+
+/**
+ * @group Form.io components
+ * @category Base types
+ */
+export interface PartnerDetails {
+  bsn?: string;
+  initials?: string;
+  affixes?: string;
+  lastName?: string;
+  dateOfBirth?: string;
+}
+/**
+ * @group Form.io components
+ * @category Concrete types
+ */
+export interface PartnersComponentSchema extends TableViewComponentSchema<PartnerDetails> {
+  type: 'partners';
+  populateFrom: string;
+}
```

Comment:
especially for the first type (and we only expect one additional type for children), I'd avoid defining a generic base interface (rule of three). I'd probably just define the entire component definition here in one go without extending/inheritance